src/operators/AMRStitchCell/PyAMRStitchCellAttributes.C

```diff
@@ -103,7 +103,7 @@ AMRStitchCellAttributes_SetCreateCellsOfType(PyObject *self, PyObject *args)
     long val = PyLong_AsLong(args);
     int cval = int(val);
 
-    if ((val == -1 && PyErr_Occurred()) || cval != val)
+    if ((val == -1 && PyErr_Occurred()) || long(cval) != val)
```

Comment:
> @markcmiller86 it looks like your latest regen didn't regenerate all the files. I noticed PyAMRStitchCellAttributes .C didn't get the latest error checking. Haven't looked further yet.

Yes it did. Right here.

Reply:
Guess I'm confused why, when attempting to parse an expected int, some code has this:
https://github.com/visit-dav/visit/blob/b6176083e0d4fdf040ae7e50489682b36e0717c1/src/operators/AxisAlignedSlice4D/PyAxisAlignedSlice4DAttributes.C#L129-L137
and others this:
https://github.com/visit-dav/visit/blob/b6176083e0d4fdf040ae7e50489682b36e0717c1/src/operators/AMRStitchCell/PyAMRStitchCellAttributes.C#L103-L110

Reply:
It has to do with whether the code was macro-ized or not. I think having a look at structure of `GeneratePython.h` would illuminate a number of issues for you. I macroized what made sense to there. In other cases, there is still custom code. Have a look a [this issue](https://github.com/visit-dav/visit/issues/5875) also. Bottom line, enums (AMRStichCell example you cite) are handled one way and general ints or floats or whatever are handled another.

Reply:
IMHO the extra checking added due to use of macro-ization doesn't make sense if not used everywhere. If the simpler error checking on parsing an int is good enough for enum, why not for general int? I do understand your desire to reduce code duplication within the Generator, but I think it makes the code harder to understand in this instance. Definitely makes reviewing harder when it appears there are two separate code paths for ints. It isn't obvious unless one checks the .xml file that you are really parsing for an enum-int not a regular-int.

Reply:
IMHO, if you look at [this issue](https://github.com/visit-dav/visit/issues/5875), the problems you are pointing to existed *before* my changes here. My changes here *reduce* duplication and *increase* error checking and IMHO certainly do not make the code any harder to understand.

I think the problem you are having is an expectation of either simplicity and consistency that did not exist before and still exists now though to a far lesser extent.

I believe I can fully *defend* all of these changes with examples of where our python CLI accepts bad inputs and does something bad or strange and/or with where we reduce code. So, I am sorry, I do not agree with your objections in this example. Bottom line, the macro-ization I added cannot be *easily* added *everywhere* and the belief or expectation that it is is unreasnable.

Reply:
BTW....a slew of tests of both and good inputs are tested in `atts_assign.py` in this PR. I challenge you to find examples of cases in there that *should* work but don't or that don't work but should...

```
    BEGIN SECTION: Assignment to bool member, "inverse", (of CylinderAttributes())
    Test case 'ca.inverse=1,2' (eq) PASSED
    Test case 'ca.inverse='123'' (eq) PASSED
    Test case 'ca.inverse=(1+2j)' (eq) PASSED
    Test case 'ca.inverse=[2, 4, 6]' (eq) PASSED
    Test case 'ca.inverse=None' (eq) PASSED
    Test case 'ca.inverse=5' (eq) PASSED
    Test case 'ca.SetInverse('123')' (eq) PASSED
    Test case 'ca.SetInverse((1+2j))' (eq) PASSED
    Test case 'ca.SetInverse([2, 4, 6])' (eq) PASSED
    Test case 'ca.SetInverse(None)' (eq) PASSED
    Test case 'ca.SetInverse(5)' (eq) PASSED
    Test case 'ca.inverse=0' (eq) PASSED
    Test case 'ca.inverse=1' (eq) PASSED
    Test case 'ca.inverse=True' (eq) PASSED
    Test case 'ca.inverse=False' (eq) PASSED
    Test case 'ca.SetInverse(0)' (eq) PASSED
    Test case 'ca.SetInverse(1)' (eq) PASSED
    Test case 'ca.SetInverse(True)' (eq) PASSED
    Test case 'ca.SetInverse(False)' (eq) PASSED
    BEGIN SECTION: Assignment to int member, "samplesPerRay", (of VolumeAttributes())
    Test case 'va.samplesPerRay=1,2' (eq) PASSED
    Test case 'va.samplesPerRay='123'' (eq) PASSED
    Test case 'va.samplesPerRay=(1+2j)' (eq) PASSED
    Test case 'va.samplesPerRay=None' (eq) PASSED
    Test case 'va.samplesPerRay=[2, 4, 6]' (eq) PASSED
    Test case 'va.samplesPerRay=2147483648' (eq) PASSED
    Test case 'va.SetSamplesPerRay('123')' (eq) PASSED
    Test case 'va.SetSamplesPerRay((1+2j))' (eq) PASSED
    Test case 'va.SetSamplesPerRay(None)' (eq) PASSED
    Test case 'va.SetSamplesPerRay([2, 4, 6])' (eq) PASSED
    Test case 'va.SetSamplesPerRay(2147483648)' (eq) PASSED
    Test case 'va.samplesPerRay=0' (eq) PASSED
    Test case 'va.samplesPerRay=1' (eq) PASSED
    Test case 'va.samplesPerRay=-1' (eq) PASSED
    Test case 'va.samplesPerRay=5' (eq) PASSED
    Test case 'va.samplesPerRay=True' (eq) PASSED
    Test case 'va.samplesPerRay=False' (eq) PASSED
    Test case 'va.samplesPerRay=2147483647' (eq) PASSED
    Test case 'va.SetSamplesPerRay(0)' (eq) PASSED
    Test case 'va.SetSamplesPerRay(1)' (eq) PASSED
    Test case 'va.SetSamplesPerRay(-1)' (eq) PASSED
    Test case 'va.SetSamplesPerRay(5)' (eq) PASSED
    Test case 'va.SetSamplesPerRay(True)' (eq) PASSED
    Test case 'va.SetSamplesPerRay(False)' (eq) PASSED
    Test case 'va.SetSamplesPerRay(2147483647)' (eq) PASSED
    BEGIN SECTION: Assignment to float member, "opacityAttenuation", (of VolumeAttributes())
    Test case 'va.opacityAttenuation=1,2' (eq) PASSED
    Test case 'va.opacityAttenuation='123'' (eq) PASSED
    Test case 'va.opacityAttenuation=(1+2j)' (eq) PASSED
    Test case 'va.opacityAttenuation=None' (eq) PASSED
    Test case 'va.opacityAttenuation=[2, 4, 6]' (eq) PASSED
    Test case 'va.opacityAttenuation=3.402823e+39' (eq) PASSED
    Test case 'va.SetOpacityAttenuation('123')' (eq) PASSED
    Test case 'va.SetOpacityAttenuation((1+2j))' (eq) PASSED
    Test case 'va.SetOpacityAttenuation(None)' (eq) PASSED
    Test case 'va.SetOpacityAttenuation([2, 4, 6])' (eq) PASSED
    Test case 'va.SetOpacityAttenuation(3.402823e+39)' (eq) PASSED
    Test case 'va.opacityAttenuation=0' (eq) PASSED
    Test case 'va.opacityAttenuation=1' (eq) PASSED
    Test case 'va.opacityAttenuation=-1' (eq) PASSED
    Test case 'va.opacityAttenuation=0.3' (eq) PASSED
    Test case 'va.opacityAttenuation=3.40282e+37' (eq) PASSED
    Test case 'va.opacityAttenuation=True' (eq) PASSED
    Test case 'va.opacityAttenuation=False' (eq) PASSED
    Test case 'va.SetOpacityAttenuation(0)' (eq) PASSED
    Test case 'va.SetOpacityAttenuation(1)' (eq) PASSED
    Test case 'va.SetOpacityAttenuation(-1)' (eq) PASSED
    Test case 'va.SetOpacityAttenuation(0.3)' (eq) PASSED
    Test case 'va.SetOpacityAttenuation(3.40282e+37)' (eq) PASSED
    Test case 'va.SetOpacityAttenuation(True)' (eq) PASSED
    Test case 'va.SetOpacityAttenuation(False)' (eq) PASSED
    BEGIN SECTION: Assignment to double member, "radius", (of CylinderAttributes())
    Test case 'ca.radius=1,2' (eq) PASSED
    Test case 'ca.radius='123'' (eq) PASSED
    Test case 'ca.radius=(1+2j)' (eq) PASSED
    Test case 'ca.radius=None' (eq) PASSED
    Test case 'ca.radius=[2, 4, 6]' (eq) PASSED
    Test case 'ca.SetRadius('123')' (eq) PASSED
    Test case 'ca.SetRadius((1+2j))' (eq) PASSED
    Test case 'ca.SetRadius(None)' (eq) PASSED
    Test case 'ca.SetRadius([2, 4, 6])' (eq) PASSED
    Test case 'ca.radius=0' (eq) PASSED
    Test case 'ca.radius=1' (eq) PASSED
    Test case 'ca.radius=-1' (eq) PASSED
    Test case 'ca.radius=5.5' (eq) PASSED
    Test case 'ca.radius=0.0' (eq) PASSED
    Test case 'ca.radius=inf' (eq) PASSED
    Test case 'ca.radius=True' (eq) PASSED
    Test case 'ca.radius=False' (eq) PASSED
    Test case 'ca.SetRadius(0)' (eq) PASSED
    Test case 'ca.SetRadius(1)' (eq) PASSED
    Test case 'ca.SetRadius(-1)' (eq) PASSED
    Test case 'ca.SetRadius(5.5)' (eq) PASSED
    Test case 'ca.SetRadius(0.0)' (eq) PASSED
    Test case 'ca.SetRadius(inf)' (eq) PASSED
    Test case 'ca.SetRadius(True)' (eq) PASSED
    Test case 'ca.SetRadius(False)' (eq) PASSED
    BEGIN SECTION: Assignment to string member, "designator", (of CurveAttributes())
    Test case 'ca.designator="123","abc"' (eq) PASSED
    Test case 'ca.designator=0' (eq) PASSED
    Test case 'ca.designator=1' (eq) PASSED
    Test case 'ca.designator=1.1' (eq) PASSED
    Test case 'ca.designator=(1+2j)' (eq) PASSED
    Test case 'ca.designator=None' (eq) PASSED
    Test case 'ca.designator=[2, 4, 6]' (eq) PASSED
    Test case 'ca.SetDesignator(0)' (eq) PASSED
    Test case 'ca.SetDesignator(1)' (eq) PASSED
    Test case 'ca.SetDesignator(1.1)' (eq) PASSED
    Test case 'ca.SetDesignator((1+2j))' (eq) PASSED
    Test case 'ca.SetDesignator(None)' (eq) PASSED
    Test case 'ca.SetDesignator([2, 4, 6])' (eq) PASSED
    Test case 'ca.designator='123'' (eq) PASSED
    Test case 'ca.designator='abc'' (eq) PASSED
    Test case 'ca.designator=''' (eq) PASSED
    Test case 'ca.SetDesignator('123')' (eq) PASSED
    Test case 'ca.SetDesignator('abc')' (eq) PASSED
    Test case 'ca.SetDesignator('')' (eq) PASSED
    BEGIN SECTION: Assignment to GlyphType member, "pointType", (of MeshAttributes())
    Test case 'ma.pointType=1' (eq) PASSED
    Test case 'ma.pointType='123'' (eq) PASSED
    Test case 'ma.pointType=(1+2j)' (eq) PASSED
    Test case 'ma.pointType=None' (eq) PASSED
    Test case 'ma.pointType=[2, 4, 6]' (eq) PASSED
    Test case 'ma.pointType=-1' (eq) PASSED
    Test case 'ma.pointType=123123123123123123123123123123' (eq) PASSED
    Test case 'ma.SetPointType('123')' (eq) PASSED
    Test case 'ma.SetPointType((1+2j))' (eq) PASSED
    Test case 'ma.SetPointType(None)' (eq) PASSED
    Test case 'ma.SetPointType([2, 4, 6])' (eq) PASSED
    Test case 'ma.SetPointType(-1)' (eq) PASSED
    Test case 'ma.SetPointType(123123123123123123123123123123)' (eq) PASSED
    Test case 'ma.pointType=0' (eq) PASSED
    Test case 'ma.pointType=1' (eq) PASSED
    Test case 'ma.pointType=5' (eq) PASSED
    Test case 'ma.pointType=True' (eq) PASSED
    Test case 'ma.pointType=False' (eq) PASSED
    Test case 'ma.pointType=6' (eq) PASSED
    Test case 'ma.SetPointType(0)' (eq) PASSED
    Test case 'ma.SetPointType(1)' (eq) PASSED
    Test case 'ma.SetPointType(5)' (eq) PASSED
    Test case 'ma.SetPointType(True)' (eq) PASSED
    Test case 'ma.SetPointType(False)' (eq) PASSED
    Test case 'ma.SetPointType(6)' (eq) PASSED
    BEGIN SECTION: Assignment to Enum member, "smoothingLevel", (of MeshAttributes())
    Test case 'ma.smoothingLevel=1' (eq) PASSED
    Test case 'ma.smoothingLevel='123'' (eq) PASSED
    Test case 'ma.smoothingLevel=(1+2j)' (eq) PASSED
    Test case 'ma.smoothingLevel=None' (eq) PASSED
    Test case 'ma.smoothingLevel=[2, 4, 6]' (eq) PASSED
    Test case 'ma.smoothingLevel=-1' (eq) PASSED
    Test case 'ma.smoothingLevel=123123123' (eq) PASSED
    Test case 'ma.smoothingLevel=123123123123123123123123123123' (eq) PASSED
    Test case 'ma.SetSmoothingLevel('123')' (eq) PASSED
    Test case 'ma.SetSmoothingLevel((1+2j))' (eq) PASSED
    Test case 'ma.SetSmoothingLevel(None)' (eq) PASSED
    Test case 'ma.SetSmoothingLevel([2, 4, 6])' (eq) PASSED
    Test case 'ma.SetSmoothingLevel(-1)' (eq) PASSED
    Test case 'ma.SetSmoothingLevel(123123123)' (eq) PASSED
    Test case 'ma.SetSmoothingLevel(123123123123123123123123123123)' (eq) PASSED
    Test case 'ma.smoothingLevel=0' (eq) PASSED
    Test case 'ma.smoothingLevel=1' (eq) PASSED
    Test case 'ma.smoothingLevel=2' (eq) PASSED
    Test case 'ma.smoothingLevel=True' (eq) PASSED
    Test case 'ma.smoothingLevel=False' (eq) PASSED
    Test case 'ma.smoothingLevel=1' (eq) PASSED
    Test case 'ma.SmoothingLevel(0)' (eq) PASSED
    Test case 'ma.SmoothingLevel(1)' (eq) PASSED
    Test case 'ma.SmoothingLevel(2)' (eq) PASSED
    Test case 'ma.SmoothingLevel(True)' (eq) PASSED
    Test case 'ma.SmoothingLevel(False)' (eq) PASSED
    Test case 'ma.SmoothingLevel(1)' (eq) PASSED
    BEGIN SECTION: Assignment to tuple, "point1", member (of CylinderAttributes())
    Test case 'ca.point=1,2,3' (eq) PASSED
    Test case 'ca.SetPoint(1,2,3)' (eq) PASSED
    Test case 'ca.point1=1,2' (eq) PASSED
    Test case 'ca.point1=1,2,3,4' (eq) PASSED
    Test case 'ca.point1=1, 2' (eq) PASSED
    Test case 'ca.point1=1, 2, 3, 4' (eq) PASSED
    Test case 'ca.point1='123'' (eq) PASSED
    Test case 'ca.point1=1, (1+2j), 3' (eq) PASSED
    Test case 'ca.point1=1, [2, 4, 6], 3' (eq) PASSED
    Test case 'ca.point1=1, 'b', 3' (eq) PASSED
    Test case 'ca.point1=1, None, 3' (eq) PASSED
    Test case 'ca.SetPoint1(1, 2)' (eq) PASSED
    Test case 'ca.SetPoint1(1, 2, 3, 4)' (eq) PASSED
    Test case 'ca.SetPoint1('123')' (eq) PASSED
    Test case 'ca.SetPoint1(1, (1+2j), 3)' (eq) PASSED
    Test case 'ca.SetPoint1(1, [2, 4, 6], 3)' (eq) PASSED
    Test case 'ca.SetPoint1(1, 'b', 3)' (eq) PASSED
    Test case 'ca.SetPoint1(1, None, 3)' (eq) PASSED
    Test case 'ca.point1=1,2,3' (eq) PASSED
    Test case 'ca.point1=1, 2, 3' (eq) PASSED
    Test case 'ca.point1=1.1, 2.2, 3.3' (eq) PASSED
    Test case 'ca.point1=2, 4, 6' (eq) PASSED
    Test case 'ca.SetPoint1(1, 2, 3)' (eq) PASSED
    Test case 'ca.SetPoint1(1.1, 2.2, 3.3)' (eq) PASSED
    Test case 'ca.SetPoint1(2, 4, 6)' (eq) PASSED
    BEGIN SECTION: Assignment to ucharVector member, "changedColors",  (of MultiCurveAttributes())
    Test case 'mca.changedColors=1,2,3' (eq) PASSED
    Test case 'mca.changedColors=1, 123123123123123123123123123123, 3' (eq) PASSED
    Test case 'mca.changedColors=1, (1+2j), 3' (eq) PASSED
    Test case 'mca.changedColors=1, [2, 4, 6], 3' (eq) PASSED
    Test case 'mca.changedColors=1, 'b', 3' (eq) PASSED
    Test case 'mca.changedColors=1, None, 3' (eq) PASSED
    Test case 'mca.changedColors='123',' (eq) PASSED
    Test case 'mca.SetChangedColors(1, 123123123123123123123123123123, 3)' (eq) PASSED
    Test case 'mca.SetChangedColors(1, (1+2j), 3)' (eq) PASSED
    Test case 'mca.SetChangedColors(1, [2, 4, 6], 3)' (eq) PASSED
    Test case 'mca.SetChangedColors(1, 'b', 3)' (eq) PASSED
    Test case 'mca.SetChangedColors(1, None, 3)' (eq) PASSED
    Test case 'mca.SetChangedColors('123',)' (eq) PASSED
    Test case 'mca.changedColors=1, 2, 3' (eq) PASSED
    Test case 'mca.changedColors=2, 4, 6' (eq) PASSED
    Test case 'mca.changedColors=1, True, 3' (eq) PASSED
    Test case 'mca.changedColors=1, False, 3' (eq) PASSED
    Test case 'mca.SetChangedColors(1, 2, 3)' (eq) PASSED
    Test case 'mca.SetChangedColors(2, 4, 6)' (eq) PASSED
    Test case 'mca.SetChangedColors(1, True, 3)' (eq) PASSED
    Test case 'mca.SetChangedColors(1, False, 3)' (eq) PASSED
    BEGIN SECTION: Assignment to intVector member, "index", (of OnionPeelAttributes())
    Test case 'opa.index=1,2,3' (eq) PASSED
    Test case 'opa.index=2147483648,' (eq) PASSED
    Test case 'opa.index=1+2j),' (eq) PASSED
    Test case 'opa.index='b',' (eq) PASSED
    Test case 'opa.index=None,' (eq) PASSED
    Test case 'opa.index=1, 2147483648, 3' (eq) PASSED
    Test case 'opa.index=1, (1+2j), 3' (eq) PASSED
    Test case 'opa.index=1, [2, 4, 6], 3' (eq) PASSED
    Test case 'opa.index=1, 'b', 3' (eq) PASSED
    Test case 'opa.index=1, None, 3' (eq) PASSED
    Test case 'opa.SetIndex(2147483648,)' (eq) PASSED
    Test case 'opa.SetIndex(1+2j),)' (eq) PASSED
    Test case 'opa.SetIndex('b',)' (eq) PASSED
    Test case 'opa.SetIndex(None,)' (eq) PASSED
    Test case 'opa.SetIndex(1, 2147483648, 3)' (eq) PASSED
    Test case 'opa.SetIndex(1, (1+2j), 3)' (eq) PASSED
    Test case 'opa.SetIndex(1, [2, 4, 6], 3)' (eq) PASSED
    Test case 'opa.SetIndex(1, 'b', 3)' (eq) PASSED
    Test case 'opa.SetIndex(1, None, 3)' (eq) PASSED
    Test case 'opa.index=1, 2, 3' (eq) PASSED
    Test case 'opa.index=[2, 4, 6]' (eq) PASSED
    Test case 'opa.index=2, 4, 6' (eq) PASSED
    Test case 'opa.index=1, True, 3' (eq) PASSED
    Test case 'opa.index=1, False, 3' (eq) PASSED
    Test case 'opa.index=1, 2147483647, 3' (eq) PASSED
    Test case 'opa.SetIndex(1, 2, 3)' (eq) PASSED
    Test case 'opa.SetIndex([2, 4, 6])' (eq) PASSED
    Test case 'opa.SetIndex(2, 4, 6)' (eq) PASSED
    Test case 'opa.SetIndex(1, True, 3)' (eq) PASSED
    Test case 'opa.SetIndex(1, False, 3)' (eq) PASSED
    Test case 'opa.SetIndex(1, 2147483647, 3)' (eq) PASSED
    BEGIN SECTION: Assignment to doubleVector member, "values", (of ContourAttributes())
    Test case 'ca.contourValue=1,2,3' (eq) PASSED
    Test case 'ca.contourValue=1+2j),' (eq) PASSED
    Test case 'ca.contourValue='b',' (eq) PASSED
    Test case 'ca.contourValue=None,' (eq) PASSED
    Test case 'ca.contourValue=1, (1+2j), 3' (eq) PASSED
    Test case 'ca.contourValue=1, [2, 4, 6], 3' (eq) PASSED
    Test case 'ca.contourValue=1, 'b', 3' (eq) PASSED
    Test case 'ca.contourValue=1, None, 3' (eq) PASSED
    Test case 'ca.SetContourValue(1+2j),)' (eq) PASSED
    Test case 'ca.SetContourValue('b',)' (eq) PASSED
    Test case 'ca.SetContourValue(None,)' (eq) PASSED
    Test case 'ca.SetContourValue(1, (1+2j), 3)' (eq) PASSED
    Test case 'ca.SetContourValue(1, [2, 4, 6], 3)' (eq) PASSED
    Test case 'ca.SetContourValue(1, 'b', 3)' (eq) PASSED
    Test case 'ca.SetContourValue(1, None, 3)' (eq) PASSED
    Test case 'ca.contourValue=1, 2, 3' (eq) PASSED
    Test case 'ca.contourValue=[2, 4, 6]' (eq) PASSED
    Test case 'ca.contourValue=2, 4, 6' (eq) PASSED
    Test case 'ca.contourValue=1, True, 3' (eq) PASSED
    Test case 'ca.contourValue=1, False, 3' (eq) PASSED
    Test case 'ca.SetContourValue(1, 2, 3)' (eq) PASSED
    Test case 'ca.SetContourValue([2, 4, 6])' (eq) PASSED
    Test case 'ca.SetContourValue(2, 4, 6)' (eq) PASSED
    Test case 'ca.SetContourValue(1, True, 3)' (eq) PASSED
    Test case 'ca.SetContourValue(1, False, 3)' (eq) PASSED
    BEGIN SECTION: Assignment to ucharArray member, "freeformOpacity", (of VolumeAttributes())
    Test case 'va.freeformOpacity=3,17' (eq) PASSED
    Test case 'va.SetFreeformOpacity(3,17)' (eq) PASSED
    Test case 'va.freeformOpacity=tuple(arr)' (eq) PASSED
    Test case 'va.SetFreeformOpacity(*tuple(arr))' (eq) PASSED
    Test case 'va.freeformOpacity=3,None' (eq) PASSED
    Test case 'va.freeformOpacity=3,(1+2j)' (eq) PASSED
    Test case 'va.freeformOpacity=3,[2, 4, 6]' (eq) PASSED
    Test case 'va.freeformOpacity=3,'123'' (eq) PASSED
    Test case 'va.freeformOpacity=None,17' (eq) PASSED
    Test case 'va.freeformOpacity=(1+2j),17' (eq) PASSED
    Test case 'va.freeformOpacity=[2, 4, 6],17' (eq) PASSED
    Test case 'va.freeformOpacity='123',17' (eq) PASSED
    Test case 'va.freeformOpacity=-3,17' (eq) PASSED
    Test case 'va.freeformOpacity=3,1700' (eq) PASSED
    Test case 'va.SetFreeformOpacity(3,None)' (eq) PASSED
    Test case 'va.SetFreeformOpacity(3,(1+2j))' (eq) PASSED
    Test case 'va.SetFreeformOpacity(3,[2, 4, 6])' (eq) PASSED
    Test case 'va.SetFreeformOpacity(3,'123')' (eq) PASSED
    Test case 'va.SetFreeformOpacity(None,17)' (eq) PASSED
    Test case 'va.SetFreeformOpacity((1+2j),17)' (eq) PASSED
    Test case 'va.SetFreeformOpacity([2, 4, 6],17)' (eq) PASSED
    Test case 'va.SetFreeformOpacity('123',17)' (eq) PASSED
    Test case 'va.SetFreeformOpacity(-3,17)' (eq) PASSED
    Test case 'va.SetFreeformOpacity(3,1700)' (eq) PASSED
    Test case 'va.freeformOpacity=(17,)*256' (eq) PASSED
    Test case 'va.SetFreeformOpacity((17,)*256)' (eq) PASSED
    Test case 'va.freeformOpacity=(17, 17, 17, None, 17, 17, 17, ...' (eq) PASSED
    Test case 'va.freeformOpacity=(17, 17, 17, (1+2j, ..., 17, 17, 17, ...' (eq) PASSED
    Test case 'va.freeformOpacity=(17, 17, 17, [2, 4, 6], 17, 17, 17, ...' (eq) PASSED
    Test case 'va.freeformOpacity=(17, 17, 17, (1, 2, 3, ..., 17, 17, 17, ...' (eq) PASSED
    Test case 'va.freeformOpacity=(17, 17, 17, '123', 17, 17, 17, ...' (eq) PASSED
    Test case 'va.SetFreeformOpacity(17, 17, 17, None, 17, 17, 17, ...)' (eq) PASSED
    Test case 'va.SetFreeformOpacity(17, 17, 17, (1+2j, ...), 17, 17, 17, ...)' (eq) PASSED
    Test case 'va.SetFreeformOpacity(17, 17, 17, [2, 4, 6], 17, 17, 17, ...)' (eq) PASSED
    Test case 'va.SetFreeformOpacity(17, 17, 17, (1, 2, 3, ...), 17, 17, 17, ...)' (eq) PASSED
    Test case 'va.SetFreeformOpacity(17, 17, 17, '123', 17, 17, 17, ...)' (eq) PASSED
    BEGIN SECTION: Assignment to intArray member, "reflections", (of ReflectAttributes())
    Test case 'ra.reflections=0,1,0,1,0,1,0,1' (eq) PASSED
    Test case 'ra.reflections=0, 1, None, 1, 0, 1, 0, 1' (eq) PASSED
    Test case 'ra.reflections=0, 1, (1+2j), 1, 0, 1, 0, 1' (eq) PASSED
    Test case 'ra.reflections=0, 1, [2, 4, 6], 1, 0, 1, 0, 1' (eq) PASSED
    Test case 'ra.reflections=0, 1, 2147483648, 1, 0, 1, 0, 1' (eq) PASSED
    Test case 'ra.reflections=0, 1, '123', 1, 0, 1, 0, 1' (eq) PASSED
    Test case 'ra.reflections=0, 1, 0, 1, 0, 1, 0, 1, 1' (eq) PASSED
    Test case 'ra.reflections=0, 1, 0, 1, 0, 1, 0' (eq) PASSED
    Test case 'ra.SetReflections(0, 1, None, 1, 0, 1, 0, 1)' (eq) PASSED
    Test case 'ra.SetReflections(0, 1, (1+2j), 1, 0, 1, 0, 1)' (eq) PASSED
    Test case 'ra.SetReflections(0, 1, [2, 4, 6], 1, 0, 1, 0, 1)' (eq) PASSED
    Test case 'ra.SetReflections(0, 1, 2147483648, 1, 0, 1, 0, 1)' (eq) PASSED
    Test case 'ra.SetReflections(0, 1, '123', 1, 0, 1, 0, 1)' (eq) PASSED
    Test case 'ra.SetReflections(0, 1, 0, 1, 0, 1, 0, 1, 1)' (eq) PASSED
    Test case 'ra.SetReflections(0, 1, 0, 1, 0, 1, 0)' (eq) PASSED
    Test case 'ra.reflections=0, 1, 0, 1, 0, 1, 0, 1' (eq) PASSED
    Test case 'ra.reflections=-1, 100, -1, 100, -1, 100, -1, 100' (eq) PASSED
    Test case 'ra.reflections=0, True, False, 1, 0, 1, 0, 1' (eq) PASSED
    Test case 'ra.reflections=0, 1, 2147483647, 1, 0, 1, 0, 1' (eq) PASSED
    Test case 'ra.SetReflections(0, 1, 0, 1, 0, 1, 0, 1)' (eq) PASSED
    Test case 'ra.SetReflections(-1, 100, -1, 100, -1, 100, -1, 100)' (eq) PASSED
    Test case 'ra.SetReflections(0, True, False, 1, 0, 1, 0, 1)' (eq) PASSED
    Test case 'ra.SetReflections(0, 1, 2147483647, 1, 0, 1, 0, 1)' (eq) PASSED
    BEGIN SECTION: Assignment to floatArray member, "center", (of RadialResampleAttributes())
    Test case 'rra.center=0,1,2' (eq) PASSED
    Test case 'rra.center=0,1' (eq) PASSED
    Test case 'rra.center=0,1,2,3' (eq) PASSED
    Test case 'rra.center=0, 1' (eq) PASSED
    Test case 'rra.center=0, 1, 2, 3' (eq) PASSED
    Test case 'rra.center=0, None, 2' (eq) PASSED
    Test case 'rra.center=0, (1+2j), 2' (eq) PASSED
    Test case 'rra.center=0, [2, 4, 6], 2' (eq) PASSED
    Test case 'rra.center=0, '123', 2' (eq) PASSED
    Test case 'rra.center=0, 3.402823e+39, 2' (eq) PASSED
    Test case 'rra.SetCenter(0, 1)' (eq) PASSED
    Test case 'rra.SetCenter(0, 1, 2, 3)' (eq) PASSED
    Test case 'rra.SetCenter(0, None, 2)' (eq) PASSED
    Test case 'rra.SetCenter(0, (1+2j), 2)' (eq) PASSED
    Test case 'rra.SetCenter(0, [2, 4, 6], 2)' (eq) PASSED
    Test case 'rra.SetCenter(0, '123', 2)' (eq) PASSED
    Test case 'rra.SetCenter(0, 3.402823e+39, 2)' (eq) PASSED
    Test case 'rra.center=1, 2, 3' (eq) PASSED
    Test case 'rra.center=1.1, 2.2, 3.3' (eq) PASSED
    Test case 'rra.center=2, 4, 6' (eq) PASSED
    Test case 'rra.center=1, True, 3' (eq) PASSED
    Test case 'rra.center=1, False, 3' (eq) PASSED
    Test case 'rra.center=1, 3.40282e+37, 3' (eq) PASSED
    Test case 'rra.SetCenter(1, 2, 3)' (eq) PASSED
    Test case 'rra.SetCenter(1.1, 2.2, 3.3)' (eq) PASSED
    Test case 'rra.SetCenter(2, 4, 6)' (eq) PASSED
    Test case 'rra.SetCenter(1, True, 3)' (eq) PASSED
    Test case 'rra.SetCenter(1, False, 3)' (eq) PASSED
    Test case 'rra.SetCenter(1, 3.40282e+37, 3)' (eq) PASSED
    BEGIN SECTION: Assignment to doubleArray member, "materialProperties", (of VolumeAttributes())
    Test case 'va.materialProperties=0,1,2,3' (eq) PASSED
    Test case 'va.materialProperties=0,1,2' (eq) PASSED
    Test case 'va.materialProperties=0,1,2,3,4' (eq) PASSED
    Test case 'va.materialProperties=0, 1' (eq) PASSED
    Test case 'va.materialProperties=0, 1, 2, 3, 4' (eq) PASSED
    Test case 'va.materialProperties=0, None, 2, 3' (eq) PASSED
    Test case 'va.materialProperties=0, (1+2j), 2, 3' (eq) PASSED
    Test case 'va.materialProperties=0, [2, 4, 6], 2, 3' (eq) PASSED
    Test case 'va.materialProperties=0, '123', 2, 3' (eq) PASSED
    Test case 'va.SetMaterialProperties(0, 1)' (eq) PASSED
    Test case 'va.SetMaterialProperties(0, 1, 2, 3, 4)' (eq) PASSED
    Test case 'va.SetMaterialProperties(0, None, 2, 3)' (eq) PASSED
    Test case 'va.SetMaterialProperties(0, (1+2j), 2, 3)' (eq) PASSED
    Test case 'va.SetMaterialProperties(0, [2, 4, 6], 2, 3)' (eq) PASSED
    Test case 'va.SetMaterialProperties(0, '123', 2, 3)' (eq) PASSED
    Test case 'va.materialProperties=1, 2, 3, 4' (eq) PASSED
    Test case 'va.materialProperties=1.1, 2.2, 3.3, 4.4' (eq) PASSED
    Test case 'va.materialProperties=1, True, 3, 4' (eq) PASSED
    Test case 'va.materialProperties=1, False, 3, 4' (eq) PASSED
    Test case 'va.SetMaterialProperties(1, 2, 3, 4)' (eq) PASSED
    Test case 'va.SetMaterialProperties(1.1, 2.2, 3.3, 4.4)' (eq) PASSED
    Test case 'va.SetMaterialProperties(1, True, 3, 4)' (eq) PASSED
    Test case 'va.SetMaterialProperties(1, False, 3, 4)' (eq) PASSED
    BEGIN SECTION: ColorAttribute stuff
    Test case 'ca.color=0, 1, 2' (eq) PASSED
    Test case 'ca.color=0, 1, 2, 3, 4' (eq) PASSED
    Test case 'ca.color=0, None, 2, 3' (eq) PASSED
    Test case 'ca.color=0, (1+2j), 2, 3' (eq) PASSED
    Test case 'ca.color=0, [2, 4, 6], 2, 3' (eq) PASSED
    Test case 'ca.color=0, '123', 2, 3' (eq) PASSED
    Test case 'ca.color=0, -1, 2, 3' (eq) PASSED
    Test case 'ca.color=0, 256, 2, 3' (eq) PASSED
    Test case 'ca.SetColor(0, 1, 2)' (eq) PASSED
    Test case 'ca.SetColor(0, 1, 2, 3, 4)' (eq) PASSED
    Test case 'ca.SetColor(0, None, 2, 3)' (eq) PASSED
    Test case 'ca.SetColor(0, (1+2j), 2, 3)' (eq) PASSED
    Test case 'ca.SetColor(0, [2, 4, 6], 2, 3)' (eq) PASSED
    Test case 'ca.SetColor(0, '123', 2, 3)' (eq) PASSED
    Test case 'ca.SetColor(0, -1, 2, 3)' (eq) PASSED
    Test case 'ca.SetColor(0, 256, 2, 3)' (eq) PASSED
    Test case 'cla.AddColors' (eq) PASSED
    Test case 'cla.colors' (eq) PASSED
    Test case 'cla.GetColors(0)' (eq) PASSED
    Test case 'cla.GetColors(2)' (eq) PASSED
```

Reply:
Is this logic absolutely necessary to ensure we have parsed a valid 'int'?
```
if (fabs(double(val))>1.5E-7 && fabs((double(long(cval))-double(val))/double(val))>1.5E-7) 
     return PyErr_Format(PyExc_ValueError, "number not interpretable as C++ int"); 
```
If so, my argument is that it should also be in place for an enum being interpreted as an int. If not, then why use it for an int?

Reply:
Please propose a change to this PR that a) does not involve wholesale removal of error checking I've added (Yes, I think its necessary...not to *ensure we have parsed a valid* number (in this case an int)* -- that gets tested in the lines above -- but to ensure the number we've parsed actually fits into the C/C++ data type we are assigning it to. Not all Python ints and floats are assignable to C/C++ `int` and `float`/`double`. Python supports a much wider range of values and narrower range of types than C/C++ does) b) fits within the constraints of `GeneratePython.h` design and types it manages and c) is macor-izeable in at least as many cases as I added already.

Reply:
I took a fresh look at the full text of GeneratePython.h.  👍